runtime/critical_section.h

```diff
@@ -21,6 +21,25 @@ struct CriticalSectionGuard : private vk::not_copyable {
   ~CriticalSectionGuard() noexcept { leave_critical_section(); }
 };
 
+class CriticalSectionSmartGuard : private vk::not_copyable {
+public:
+  CriticalSectionSmartGuard() noexcept { enter_critical_section(); }
+
+  void leave_critical_section() noexcept {
```

Comment:
Maybe it's worth to add similar `enter_critical_section` method to make it more smart :) ? Having only one of them looks a bit strange

Reply:
It was added in the next commit